src/AspNetCoreModuleV2/AspNetCore/src/proxymodule.cxx

```diff
@@ -147,15 +124,21 @@ ASPNET_CORE_PROXY_MODULE::OnExecuteRequestHandler(
 
         m_pApplicationInfo->ExtractApplication(&pApplication);
 
-        // make sure application is in running state	
-        // cannot recreate the application as we cannot reload clr for inprocess	
-        if (pApplication != NULL &&
-            pApplication->QueryStatus() != APPLICATION_STATUS::RUNNING &&
-            pApplication->QueryStatus() != APPLICATION_STATUS::STARTING)
+        DBG_ASSERT(pHttpContext);
+        // make sure application is in running state
+        // cannot recreate the application as we cannot reload clr for inprocess
+        if (pApplication->QueryStatus() == APPLICATION_STATUS::OFFLINE)
+        {
+            pApplicationManager->RecycleApplicationFromManager(m_pApplicationInfo->QueryConfig()->QueryConfigPath()->QueryStr());
+            // todo: add appoffline
```

Comment:
Is this Todo resolved? Doesn't look like it.

Reply:
yes. updated it